### PR TITLE
Rename defaultVisible, visible, setVisible on disclosure

### LINKED ISSUE #1426: [v2] Rename `visible`/`setVisible` to `open`/`setOpen`
### Motivation

I'm thinking about renaming the `defaultVisible`, `visible`, and `setVisible` props on disclosure and derived state hooks to `defaultOpen`, `open`, and `setOpen`.

The main motivation is that this is more aligned with the HTML spec, where these already exist:

```html
<details open>
<dialog open>
```

The second reason is to avoid confusion about what `visible` means. For example:

```jsx
const select = useSelectState({ visible: true })

return (
  <>
    <Select state={select} />
    <SelectPopover state={select} />
  </>
)
```

For a React developer reading that code for the first time, without previous knowledge of Ariakit, it may not be obvious whether `visible` is referring to the visibility of the whole select widget (including the select button) or only the popover.

With `open`, even if you consider the select button element, it still makes sense since a select button being "open" means its popover is being displayed:

```jsx
const select = useSelectState({ open: true })

return (
  <>
    <Select state={select} />
    <SelectPopover state={select} />
  </>
)
```

### Usage example

```jsx
const disclosure = useDisclosureState({ open: true })
disclosure.setOpen(false)
```

### Possible implementations

We basically have to change the implementation and types on the disclosure state and adapt the rest of the code.

### PR DESCRIPTION
**Pull request by harshilparmar**
As discussed in #1426, I have changed the name of `defaultVisible`, `visible`, and `setVisible` to `defaultOpen`, `open`, and `setOpen` respectively.
@diegohaz Can you please take a look and please let me know If something needs to be change.

Closes #1426

### CODE REVIEW

**Comment by diegohaz** [comment]
Thank you for working on this @harshilparmar! The other components that are using `visible` need to be updated.

File: packages/ariakit/src/dialog/dialog.tsx [hunk 1]
```diff
@@ -189,7 +189,7 @@ export const useDialog = createHook<DialogOptions>(
 
     // Renders a hidden dismiss button at the top of the modal dialog element.
     // So that screen reader users aren't trapped in the dialog when there's no
-    // visible dismiss button.
+    // open dismiss button.
```
Comment (diegohaz) [inline packages/ariakit/src/dialog/dialog.tsx]:
```suggestion
    // visible dismiss button.
```

File: packages/ariakit/src/dialog/dialog.tsx [hunk 2]
```diff
@@ -223,7 +223,7 @@ export const useDialog = createHook<DialogOptions>(
       if (modal) {
         return chain(
           disableAccessibilityTreeOutside(element),
-          // When the backdrop is not visible, we also need to disable pointer
+          // When the backdrop is not open, we also need to disable pointer
```
Comment (diegohaz) [inline packages/ariakit/src/dialog/dialog.tsx]:
```suggestion
          // When the backdrop is not visible, we also need to disable pointer
```

**Review by diegohaz — APPROVED** [review]
Nice! Thank you for working on this @harshilparmar!

**Comment by harshilparmar** [comment]
> Nice! Thank you for working on this @harshilparmar!

It was my pleasure.Looking forward to more contribution 🚀 Thanks!!

**Pull request merged**